pkg/apiserver/rest/asset_scan_controller.go

```diff
@@ -45,6 +45,14 @@ func (s *ServerImpl) PostAssetScans(ctx echo.Context) error {
 		return sendError(ctx, http.StatusBadRequest, fmt.Sprintf("failed to bind request: %v", err))
 	}
 
+	_, ok := assetScan.GetResourceCleanupStatus()
+	if !ok {
+		assetScan.ResourceCleanupStatus = models.NewResourceCleanupStatusWithDefaults()
+	}
```

Comment:
A pattern we've been following for the other objects is using the reconciler to set missing defaults instead of putting logic in the rest controllers, in this case the reconciler would detect ResourceCleanupStatus == nil, and then set it to Pending prior to processing anything else, the same logic can/should apply to any other defaults.
If somehow the ResourceCleanupStatus gets cleared, the reconciler should fix it.

If we're going to move this logic into the rest controllers then we needs to be consistent with handling all defaults in the rest controllers not just the states.

Reply:
Probably the best would be to move the defaulting to `Orchestrator` and the `Scanner CLI` (CICD mode) for now and let the `API Server` to return a HTTP400 if the `ResourceCleanupStatus` field is missing from the request.

Reply:
I would prefer that it wasn't a 400 error, i.e. the object gets created with just the fields that the user specifies populated, and then the orchestrator backfills any missing information as part of the first steps in the reconcile loop. The object will simply remain unchanged if the orchestrator isn't running.

Reply:
@Tehsmash What about the `CICD` mode? There is no `Orchestrator` involved in that case if I am not mistaken.

Reply:
thats a good point, in the CICD case if ResourceCleanupStatus is required (400 error), then the client should be POSTing something like `State: "Skipped", Reason:"NotRelevant"`.

Alternatively if we make it optional to we can leave it empty because its not relevant to that mode, and it'll never get populated because the orchestrator will never look at it.

Reply:
@chrisgacsal wdyt?

Reply:
> thats a good point, in the CICD case if ResourceCleanupStatus is required (400 error), then the client should be POSTing something like State: "Skipped", Reason:"NotRelevant".

Agreed. Probably I would use the `State: "Skipped", Reason:"NotApplicable"` instead. I vote for this approach.

> Alternatively if we make it optional to we can leave it empty because its not relevant to that mode, and it'll never get populated because the orchestrator will never look at it.

I prefer the former better.

@Tehsmash ^

Reply:
Sounds good, lets go with that.

Reply:
I removed the defaulting from the AssetScan resource POST endpoint, and added to the CLI.